components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/utils/InboundWebsocketProcessorUtil.java

```diff
@@ -424,7 +427,13 @@ public static boolean isAuthenticated(InboundMessageContext inboundMessageContex
                         String keyManager = ServiceReferenceHolder.getInstance().getJwtValidationService()
                                 .getKeyManagerNameIfJwtValidatorExist(inboundMessageContext.getSignedJWTInfo());
                         if (StringUtils.isNotEmpty(keyManager)) {
-                            isJwtToken = true;
+                            if (keyManagerList.contains(APIConstants.KeyManager.API_LEVEL_ALL_KEY_MANAGERS) ||
+                                    keyManagerList.contains(keyManager)) {
+                                isJwtToken = true;
+                            } else {
+                                    throw new APISecurityException(APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
```

Comment:
Can we look into giving a much more informative error message? Or even a debug log?